File: caesar/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    //check that input is valid
    if (argc !=2)
    {
        printf("Please enter a valid encryption key\n");
        return 1;
    }
    
    //obtain the key
    int k = atoi(argv[1]);
    
    //allow the user to input the plain text to be encoded
    printf("plaintext:");
    string p = get_string();
    
    printf("ciphertext:");
    //shift each letter by k places in the alphabet while preserving the original capitalisation
    for (int i=0, n=strlen(p);i<n;i++)
    {
        if (islower(p[i]))
        {
            printf("%c", (char) ((((int) p[i] - 97 + k) % 26) + 97));
        }
        else if (isupper(p[i]))
        {
            printf("%c", (char) ((((int) p[i] - 65 + k) % 26) + 65));
        }
        else 
        {
            printf("%c", p[i]);
        }
    }
    printf("\n");
    return 0;
}

File: boscof-cs50-2017-x-pset6-20170423T095251Z/caesar.py
import cs50
import sys

def main(argv):
    #check that input is valid
    if len(sys.argv)!=2:
        print("Please enter a valid encryption key")
        exit(1)
    
    #obtain the key
    k = int(sys.argv[1])
    
    #allow the user to input the plain text to be encoded
    print("plaintext:", end="")
    s = cs50.get_string()
    
    #encrypt the plain text
    encrypt_text(s, k)
    
    #return 0 if successful
    exit(0)
    
#encrypts the plain text    
def encrypt_text(s, k):
    print("ciphertext:", end="")
    #shift each letter by k places in the alphabet while preserving case sensitivity
    for c in s:
        if c.islower():
            i = ((ord(c) - 97 + k) % 26)+97
            print("{}".format(chr(i)), end="")
        elif c.isupper():
            i = ((ord(c) - 65 + k) % 26)+65
            print("{}".format(chr(i)), end="")
        else:
            print("{}".format(c), end="")
    print("")

if __name__ == "__main__":
    main(sys.argv)
    
    